session/manager.go

```diff
@@ -159,6 +160,8 @@ func (manager *Manager) Start(session *Session, consumerID identity.Identity, co
 		return
 	}
 
+	manager.clearStaleSession(consumerID, manager.currentProposal.ServiceType)
```

Comment:
Part of cleaning happens in routine, maybe everything then

Reply:
No, we need to make sure that we got list of sessions before creating new, otherwise we can clean current session too.

Reply:
@Waldz Please answer comment. Thanks.

Reply:
Oh.. i couldn't assume it from reading code and was strange to find that just some part executed in routine. 
Maybe comment here would be beneficial.

Reply:
Comment added.